src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -916,10 +917,35 @@ private void OutputListToOutputWindow(string prefix, IEnumerable<string> values)
                     continue;
                 }
 
-                if (addedReferencePaths.Add(path))
+                if (metadataReferencesToAdd.TryGetValue(path, out var existingProperties))
                 {
-                    projectContext.AddMetadataReference(path, reference.Properties);
+                    // Merge existing aliases the properties that are already there
+                    var allAliases = existingProperties.Aliases.AddRange(reference.Properties.Aliases);
+
+                    // If one is empty and the other isn't, then we need to toss the global alias in too. The other cases
+                    // are they're both empty (and this was a direct duplicate) or they're both non-empty and the merge
+                    // is OK.
+                    if ((existingProperties.Aliases.IsDefaultOrEmpty && !reference.Properties.Aliases.IsDefaultOrEmpty) ||
+                        (!existingProperties.Aliases.IsDefaultOrEmpty && reference.Properties.Aliases.IsDefaultOrEmpty))
+                    {
+                        allAliases = allAliases.Add(MetadataReferenceProperties.GlobalAlias);
+                    }
+                    else
+                    {
+                        allAliases = allAliases.Distinct().ToImmutableArray();
```

Comment:
The ".ToImmutableArray" call is redundant, according to an analyzer that is failing your builds.